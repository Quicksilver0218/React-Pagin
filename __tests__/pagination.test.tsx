import { create } from 'react-test-renderer';
import Pagination from '../src';

test('Hide Ellipsis', () => {
  expect(
    create(
      <Pagination
        currentPage={1}
        pageCount={11}
        edgeItemCount={1}
        middleItemRange={3}
        parentElement={({children}) => (<>[{children}]</>)}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={7}
        pageCount={13}
        edgeItemCount={1}
        middleItemRange={3}
        parentElement={({children}) => (<>[{children}]</>)}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={2}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

test('1 Ellipsis', () => {
  expect(
    create(
      <Pagination
        currentPage={1}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        parentElement={({children}) => (<>[{children}]</>)}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={6}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        parentElement={({children}) => (<>[{children}]</>)}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={7}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        parentElement={({children}) => (<>[{children}]</>)}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={12}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        parentElement={({children}) => (<>[{children}]</>)}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

test('2 Ellipsis', () => {
  expect(
    create(
      <Pagination
        currentPage={7}
        pageCount={13}
        edgeItemCount={1}
        middleItemRange={3}
        parentElement={({children}) => (<>[{children}]</>)}
        firstElement="first"
        lastElement="last"
        previousElement="prev"
        nextElement="next"
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

test('All Default', () => {
  expect(
    create(
      <Pagination
        currentPage={7}
        pageCount={13}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={5}
        pageCount={10}
        pageElement={({page}) => <>{page}</>}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

test('Default Ellipsis Size', () => {
  expect(
    create(
      <Pagination
        currentPage={1}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={3}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={6}
        pageCount={11}
        edgeItemCount={1}
        middleItemRange={3}
        pageElement={({page}) => <>{page}</>}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

test('Middle Item Range', () => {
  expect(
    create(
      <Pagination
        currentPage={6}
        pageCount={11}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={5}
        pageCount={10}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={6}
        pageCount={10}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={1}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

test('Ellipsis Size', () => {
  expect(
    create(
      <Pagination
        currentPage={6}
        pageCount={11}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={2}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={6}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={2}
      />
    ).toJSON()
  ).toMatchSnapshot();

  expect(
    create(
      <Pagination
        currentPage={7}
        pageCount={12}
        edgeItemCount={1}
        middleItemRange={2}
        pageElement={({page}) => <>{page}</>}
        ellipsisElement={({side}) => <>{side}</>}
        ellipsisSize={2}
      />
    ).toJSON()
  ).toMatchSnapshot();
});