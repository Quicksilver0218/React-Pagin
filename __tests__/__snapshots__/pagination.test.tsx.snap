// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Ellipsis 1`] = `
Array [
  "[",
  "first",
  "prev",
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "end",
  "12",
  "next",
  "last",
  "]",
]
`;

exports[`1 Ellipsis 2`] = `
Array [
  "[",
  "first",
  "prev",
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "end",
  "12",
  "next",
  "last",
  "]",
]
`;

exports[`1 Ellipsis 3`] = `
Array [
  "[",
  "first",
  "prev",
  "1",
  "start",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
  "next",
  "last",
  "]",
]
`;

exports[`1 Ellipsis 4`] = `
Array [
  "[",
  "first",
  "prev",
  "1",
  "start",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
  "next",
  "last",
  "]",
]
`;

exports[`2 Ellipsis 1`] = `
Array [
  "[",
  "first",
  "prev",
  "1",
  "start",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "end",
  "13",
  "next",
  "last",
  "]",
]
`;

exports[`All Default 1`] = `
Array [
  "1",
  "start",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "end",
  "13",
]
`;

exports[`All Default 2`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "10",
]
`;

exports[`Default Ellipsis Size 1`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "end",
  "12",
]
`;

exports[`Default Ellipsis Size 2`] = `
Array [
  "1",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "11",
]
`;

exports[`Ellipsis Size 1`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
]
`;

exports[`Ellipsis Size 2`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "end",
  "12",
]
`;

exports[`Ellipsis Size 3`] = `
Array [
  "1",
  "start",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
]
`;

exports[`Hide Ellipsis 1`] = `
Array [
  "[",
  "first",
  "prev",
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "next",
  "last",
  "]",
]
`;

exports[`Hide Ellipsis 2`] = `
Array [
  "[",
  "first",
  "prev",
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
  "13",
  "next",
  "last",
  "]",
]
`;

exports[`Middle Item Range 1`] = `
Array [
  "1",
  "start",
  "4",
  "5",
  "6",
  "7",
  "8",
  "end",
  "11",
]
`;

exports[`Middle Item Range 2`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "end",
  "10",
]
`;

exports[`Middle Item Range 3`] = `
Array [
  "1",
  "start",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
]
`;
